src/lifecycle/properties.js

```diff
@@ -77,7 +77,7 @@ function property (elem, name, prop) {
 
       if (isBoolean && internalValue) {
         this.setAttribute(prop.attr, '');
-      } else if (internalValue == null || isBoolean && !internalValue) {
+      } else if (value == null || isBoolean && !internalValue) {
```

Comment:
I'm not sure if we should be doing this as far down as we are here. What this means is that the value is still being coerced and passed to the setter. This could cause the attribute to be removed even if the coercing function made it a valid value.